pkg/nodejs/yarn/parse.go

```diff
@@ -196,7 +204,9 @@ func parseBlock(block []byte, lineNum int) (lib Library, deps []string, newLine
 		if name, protocol, patterns, patternErr := parsePackagePatterns(line); patternErr == nil {
 			if patterns == nil || !validProtocol(protocol) {
 				skipBlock = true
-				err = xerrors.Errorf("failed to parse package patterns")
+				if !ignoreProtocol(protocol) {
+					err = xerrors.Errorf("failed to parse package patterns")
```

Comment:
If we don't want to fail on this error, we can just show warnings.

```suggestion
					log.Logger.Warnf("Unsupported protocol: %s", protocol)
```

If we want to fail on this error, we should return an error here.

```suggestion
					return Library{}, nil, -1,  xerrors.Errorf("failed to parse package patterns")
```

Reply:
I lean towards failing to keep it compatible with what we had before the refactor